### PR TITLE
8.next: Cake5 compatiblity

### PR DESCRIPTION
**Pull request by LordSimal**
This PR provides compatibility with the current Cake5 beta-1

This also of course required the following base adjustments in the composer.json
```
    "require": {
        "php": ">=8.1.0",
        "cakephp/cakephp": "5.x-dev"
    },
    "require-dev": {
        "cakephp/authorization": "3.x-dev",
        "cakephp/cakephp-codesniffer": "^5.0",
        "cakephp/authentication": "3.x-dev",
    }
```

This also updates PHPStan and Psalm to the latest versions as well as PHPUnit to 9.5

I first tried to make CakeDC/Users Cake5 compatible but saw, that this plugin needs to be updated beforehand.

Since the current main branch is `6.next-cake4` I first did all the adjustments based on that one.
Unfortunately I realized pretty late that `8.next-cake4` was present as well.

Therefore I just merged 8.next into mine and fixed all the merge conflicts afterwards.

### CODE REVIEW

File: config/bootstrap.php [hunk 1]
```diff
@@ -9,17 +9,5 @@
...
-if (is_array($oauthPath)) {
-    $builder = Router::createRouteBuilder('/');
-    $builder->scope('/auth', function ($routes) use ($oauthPath) {
-        $routes->connect(
-            '/:provider',
-            $oauthPath,
-            ['provider' => implode('|', array_keys(Configure::read('OAuth.providers')))]
-        );
-    });
-}
```
Comment (LordSimal) [inline config/bootstrap.php:25]: This has been moved to the `routes()` method of the Plugin class

File: src/Authentication/DefaultWebauthn2FAuthenticationChecker.php [hunk 2]
```diff
@@ -19,25 +19,25 @@
  *
  * @package CakeDC\Auth\Auth
  */
-class DefaultWebauthn2fAuthenticationChecker implements Webauthn2fAuthenticationCheckerInterface
```
Comment (LordSimal) [inline src/Authentication/DefaultWebauthn2FAuthenticationChecker.php:22]: This was just a renamed to match the filename

**Comment by LordSimal** [comment]
due to the fact that I only checked `compsoer stan` but not `composer analyse` I only fixed phpstan issues first.
This state now is as clean as I can get it right now without digging deeper.

**Comment by rochamarcelo** [comment]
@steinkel Should this be in a new branch n.next-cake5 or are we going to renamte `8.next-cake4` to `8.next.cake5`

PS: I don't see tags for 8.x

**Comment by rochamarcelo** [comment]
Using this branch at https://github.com/CakeDC/users/pull/1015 [WIP]

**Comment by steinkel** [comment]
I'd say tagging 8.x as it is now and using a 9.next-cake5 would be the way to go, thanks!

**Pull request closed without merging**